hypnotoad/gui/gui.py

```diff
@@ -168,17 +170,22 @@ def save_options_as(self):
 
     def update_options_form(self):
         """Update the widget values in the options form, based on the current
-        values in the options dict
+        values in the options object
 
         """
 
         self.options_form.setSortingEnabled(False)
         self.options_form.cellChanged.disconnect(self.options_form_changed)
-        self.options_form.setRowCount(len(self.options) + 1)
+        self.options_form.setRowCount(len(self.options))
```

Comment:
The `+ 1` here was to leave an empty row at the bottom so the user can add a new option easily. Sorry, I should've left a comment!

One for the todo list: add an action to right-click menu to add/remove/reset individual options

Reply:
I did remove this deliberately, because I don't think it makes sense for the user to be able to add new options. All the available options have to be in `TokamakEquilibrium.default_options`, so extra user-added options couldn't affect anything. If we support multiple `Equilibrium` subclasses, we'd have to get `self.options` from different places, but I think options available in the widget still shouldn't be user-modifiable.

Reply:
Ok that seems reasonable. I wasn't sure if all available options were in one place or not.